From 503d25d683a06ba6a8a8e0d6c50452d406fe2c02 Mon Sep 17 00:00:00 2001
From: Johnny Walker <jwalker@ashley-martin.com>
Date: Mon, 21 Jul 2014 15:50:11 -0600
Subject: [PATCH 4/7] use third-party sqlite3 if pkg-config reports version
 3.8.5 installed  - not sure of the actual incompatibility, but HHVM fails to
 initialize the central repository with sqlite3 version 3.8.5

---
 CMake/FindLibSQLite.cmake | 22 +++++++++++++++++-----
 1 file changed, 17 insertions(+), 5 deletions(-)

diff --git a/CMake/FindLibSQLite.cmake b/CMake/FindLibSQLite.cmake
index 8940a96..0297a95 100644
--- a/CMake/FindLibSQLite.cmake
+++ b/CMake/FindLibSQLite.cmake
@@ -8,12 +8,24 @@
 find_package(PkgConfig)
 pkg_check_modules(PC_SQLITE3 QUIET sqlite3)
 
-find_path(LIBSQLITE3_INCLUDE_DIR
-    NAMES sqlite3.h
-    HINTS ${PC_SQLITE3_INCLUDE_DIRS})
+if (PC_SQLITE3_FOUND)
+    if (PC_SQLITE3_VERSION VERSION_EQUAL "3.8.5")
+        set(LIBSQLITE3_FOUND 0)
+        set(LIBSQLITE3_INCLUDE_DIR "LIBSQLITE3_INCLUDE_DIR-NOTFOUND")
+        set(LIBSQLITE3_LIBRARY "LIBSQLITE3_LIBRARY-NOTFOUND")
+    else()
+        find_path(LIBSQLITE3_INCLUDE_DIR
+            NAMES sqlite3.h
+            HINTS ${PC_SQLITE3_INCLUDE_DIRS})
 
-find_library(LIBSQLITE3_LIBRARY
-    NAMES sqlite3)
+        find_library(LIBSQLITE3_LIBRARY
+            NAMES sqlite3)
+    endif()
+else()
+    set(LIBSQLITE3_FOUND 0)
+    set(LIBSQLITE3_INCLUDE_DIR "LIBSQLITE3_INCLUDE_DIR-NOTFOUND")
+    set(LIBSQLITE3_LIBRARY "LIBSQLITE3_LIBRARY-NOTFOUND")
+endif (PC_SQLITE3_FOUND)
 
 include(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(
-- 
1.9.3

